cmd/main.go

```diff
@@ -32,16 +33,28 @@ import (
 
 var mode string
 var logLevel string
+var initMode bool
 
 const (
 	defaultMode     = "standalone"
 	defaultLogLevel = "INFO"
+	defaultInitMode = false
 )
 
 // Main is the entry point of helper controller
 func Main() int {
 	log.SetFormatter(&log.JSONFormatter{})
 
+	init := viper.GetBool("initmode")
+	if init == true {
+		log.Info("Starting in Init mode")
+		if err := sysprom.SetupSystemPrometheus(); err != nil {
+			log.Fatal(err, "while deploying system prometheus")
+		}
+
+		return 0
```

Comment:
We are not going the initContainer way, in case the flag --initmode=true is passed monhelper will create system prometheus objects. Mugdha has added a check:
- to wait if the CRDs and operator pods are not yet up
- if system prometheus pods have been previously created skip
- in case of error while creating log error and move ahead. If we return an error here the monhelper pod will simply keep on restarting if the cause persists.